gpt_engineer/benchmark/run.py

```diff
@@ -72,13 +86,17 @@ def run(
         task_results.append(
             TaskResult(
                 task_name=task.name,
-                assertion_results={
-                    assertion_name: assertion(exec_result)
-                    for assertion_name, assertion in task.assertions.items()
-                },
+                assertion_results=[
```

Comment:
Same here, we should be able to revert, now that we are using the AppsAssertion design. Each apps assertion will be an entry in the dictionary.